daml/Marketplace/Custody.daml

```diff
@@ -120,6 +138,22 @@ template Service
           TransferDepositRequest{..} <- fetchAndArchive transferDepositRequestCid
           exerciseByKey @AssetSettlementRule accountId AssetSettlement_Transfer with ..
 
+      nonconsuming Lifecycle : [ContractId AssetDeposit]
+        with
+          lifecycleRequestCid : ContractId LifecycleRequest
+          safekeepingDepositCid: ContractId AssetDeposit
+          fixings: Map Text (Map Date Decimal) --TODO these should be looked up from a contract on-ledger, not provided by the custodian!
+        do
+          exercise lifecycleRequestCid (Lifecycle' with investor = customer, ..)
+       
+
+-- TODO: remove when @fixings@ param above is removed; or replace with GenMap
+instance MapKey Date where
+  keyToText dt = case toGregorian dt of (y,m,d) -> unwords . intersperse "-" $ show <$> [y, fromEnum m, d]
+  keyFromText s = case mapA parseInt $ splitOn "-" s of
+                    Some [y, m, d] -> fromGregorian (y, toEnum m, d)
+                    _ -> error "invalid date input"
```

Comment:
Add the invalid key to the error output; I know the chance of this erroring is slim since its set by the `keyToText` but it does raise an error for whatever reason, then it would be good to have the input.

Reply:
👍